Activity Panel: Inbox: When the user updates WooCommerce, add a note alerting that the store notice setting has moved

allendav opened the pull request:
See #176 

Hooks into WooCommerce update. On updating WooCommerce, adds a "store notice setting has moved" note to the merchant's inbox if one isn't already in their inbox.

### Screenshots

<img width="604" alt="screen shot 2018-10-23 at 10 31 28 am" src="https://user-images.githubusercontent.com/1595739/47379290-547e8a00-d6af-11e8-94a8-e003bd2de18b.png">

### Detailed test instructions:

- To avoid having to update WooCommerce (which will be a pain), instead, modify class-wc-install.php in woocommerce as follows. (Note: check_version is called on every WooCommerce request.)
- Then, visit `/wp-admin/admin.php?page=wc-admin#/` and look in your inbox.
- Note: All actions will be enabled in a subsequent PR.

```
	/**
	 * Check WooCommerce version and run the updater is required.
	 *
	 * This check is done on all requests and runs if the versions do not match.
	 */
	public static function check_version() {
		//if ( ! defined( 'IFRAME_REQUEST' ) && version_compare( get_option( 'woocommerce_version' ), WC()->version, '<' ) ) {
		//	self::install();
			do_action( 'woocommerce_updated' );
		//}
	}
```

Note: I also DRYd includes/class-wc-admin-notes-new-sales-record.php a little in this PR.

Commit:
On update, add a note alerting that the store notice setting has moved

Commit:
Remove inadvertently added blank line

timmyc reviewed (approved):
One minor comment on link target, but testing great for me! Pre-approving.

allendav commented on includes/class-wc-admin-notes-settings-notes.php:
```diff
@@ -0,0 +1,53 @@
+		$note->add_action(
+			'open-customizer',
+			__( 'Open Customizer', 'wc-admin' ),
+			'?page=wc-admin'
```
LOL yep

timmyc replied:
:trollface:

Commit:
Go ahead and 1) handle the variety of queries that can be used
as action button links and 2) enable the button (tracking will
come next). Also had to tweak CSS for activity panel item buttons
a little

allendav commented:
@timmyc - thanks for catching that the action URL was incorrect - I have fixed that (I completely forgot to circle back to that) - in the process I went ahead and enabled the buttons to make testing easier - they just aren't tracked yet

Please give this another spin (same testing instructions as above - but first use the Test Bench Tool to delete all old notices since they have the old "URL" and will not be regenerated otherwise - https://github.com/automattic/wc-admin-notes-test-bench )

@LevinMedia - you'll see a CSS change in here - without it the button label is white and pretty unreadable. Please let me know if I you would rather a different value for the label color.

LevinMedia commented:
Heya @allendav thanks for the ping on this! I cross referenced the color with `.wp-core-ui .button, .wp-core-ui .button-secondary` which appears elsewhere in wp-admin, and that's using `#555` for it's text. 

<img width="579" alt="screen shot 2018-10-25 at 8 43 56 am" src="https://user-images.githubusercontent.com/4500952/47513233-a8fe4280-d832-11e8-9903-9b48c19ea0cc.png">

Would it be possible to use the button directly from core in this component to keep things consistent?

allendav commented:
@LevinMedia - changing the component classes to match core's would probably make this a little brittle - if core changes styles it could have unexpected visual impact - so... I'll copy the #555555 to our style sheets instead
